File: robocode2-server/src/main/java/net/robocode2/model/events/ScannedBotEvent.java
package net.robocode2.model.events;

import lombok.Value;
import net.robocode2.model.Point;

/**
 * Event sent when a bot got scanned.
 * 
 * @author Flemming N. Larsen
 */
@Value
public class ScannedBotEvent implements Event {

	/** Bot id of the bot that scanned the bot */
	int scannedByBotId;

	/** Bot id of the bot that got scanned */
	int scannedBotId;
	
	/** Energy level of the scanned bot */
	double energy;
	
	/** Position of the scanned bot */
	Point position;

	/** Driving direction of the scanned bot */
	double direction;
	
	/** Speed of the scanned bot */
	double speed;
}

File: robocode2-gui/src/main/kotlin/net/robocode2/gui/model/event/ScannedBotEvent.kt
package net.robocode2.gui.model.event

import net.robocode2.gui.model.comm.Content
import net.robocode2.gui.model.comm.ContentType

class ScannedBotEvent(
        val scannedByBotId: Int,
        val scannedBotId: Int,
        val energy: Double,
        val x: Double,
        val y: Double,
        val direction: Double,
        val speed: Double
) : Content(type = ContentType.SCANNED_BOT_EVENT.type)
